### DISCUSSION TITLE
Bash shell stop working on macOS Ventura 13.1 after brew update

### DISCUSSION THREAD

**Discussion by tuffwes** [JanDeDobbeleer/oh-my-posh#3429] [Troubleshoot] [answered]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

I've been using oh-my-posh with bash for a long time and I always upgrade without errors. For some reason today after running brew update & upgrade which also upgrades oh-my-posh, I am now getting an error after opening bash prompt.

-bash: eval: line 27: conditional binary operator expected
-bash: eval: line 27: syntax error near `MC_SID'
-bash: eval: line 27: `    if [[ -v MC_SID ]];then'

I am using an exported theme .my_theme.omp.json 

I am loading from bash as usual in .bash_profile: eval "$(oh-my-posh init bash --config ~/.my-theme.omp.json)"

I tried to go back to a regular theme like: eval "$(oh-my-posh init bash --config ~/jandedobbeleer.omp.json)" I still get the same error.  To recap, everything was working before brew upgrade just fine. I didn't change anything.

### Theme

hul10.omp.json but exported as a custom my_hul10.omp.json

### What OS are you seeing the problem on?

macOS

### Which shell are you using?

bash

### Log output

```Shell
-bash: eval: line 27: conditional binary operator expected
-bash: eval: line 27: syntax error near `MC_SID'
-bash: eval: line 27: `    if [[ -v MC_SID ]];then'
```

**Comment by lewis-yeung** [comment]
What is your Bash version?

**Comment by JanDeDobbeleer** [comment, marked as answer]
@tuffwes as @lewis-yeung stated, this is most likely due to an outdated bash version. Apple is known to ship older versions, and I'm on the same device using bash `5.2.12` without issues.

EDIT: According to the [changelog](https://github.com/bminor/bash/blob/ec8113b9861375e4e17b3307372569d429dec814/NEWS#L966) this has been part of the bash 4.2 release. Also confirmed bash on macOS is `3.2.57`, and will NOT be updated by Apple ever.

```
GNU bash, version 3.2.57(1)-release (arm64-apple-darwin22)
Copyright (C) 2007 Free Software Foundation, Inc.
```

You can update bash using [homebrew](https://brew.sh/) `brew install bash`, make sure that you **prepend** homebrew's excutables path to your `PATH`.

**Reply by tuffwes** [reply]
As @lewis-yeung pointed out I was indeed using an older version of bash. Upgraded bash and the error is gone. Thank you.

**Comment by oalders** [comment]
I've got an updated `bash`, but I still need to `exec bash` after opening a new tab.

<img width="700" alt="Screenshot 2023-02-08 at 3 57 56 PM" src="https://user-images.githubusercontent.com/96205/217649589-9a72068c-ea43-4e67-a82a-cf74321ba2e4.png">

**Reply by JanDeDobbeleer** [reply]
@oalders that error at the top doesn't seem like it's an updated bash. Otherwise exec bash wouldn't work either. Are you certain the bash shell that launches isn't the old shell (hardcoded to /bin/bash for example) and running exec bash executes the correct, updated bash shell?

**Reply by oalders** [reply]
You are correct!

```bash
olafs-imac:dot-files olaf$ echo $BASH_VERSION
3.2.57(1)-release
olafs-imac:dot-files olaf$ which bash
/opt/homebrew/bin/bash
olafs-imac:dot-files olaf$ which -a bash
/opt/homebrew/bin/bash
/bin/bash
olafs-imac:dot-files olaf$ /opt/homebrew/bin/bash --version
GNU bash, version 5.2.15(1)-release (aarch64-apple-darwin22.1.0)
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

I had to add `/opt/homebrew/bin/bash` to `/etc/shells` and then `chsh -s /opt/homebrew/bin/bash $USER`.

Now, I'm all set. Thanks very much for your help @JanDeDobbeleer! 🚀 

See also https://apple.stackexchange.com/a/232983

**Reply by iandunn** [reply]
This fixed it for me, thanks!

I've opened #6760 to document it.
